make-prs.py

```diff
@@ -0,0 +1,251 @@
+            exit(1)
+
+
+def commit_diff(head: str):
+    logger.info("Pushing changes ...")
+
+    add = ["git", "add", "."]
+    subprocess.run(add, cwd=WORKING_DIR, capture_output=True, text=True)
+
+    commit = ["git", "commit", "-m", "chore: merging diff"]
+    subprocess.run(commit, cwd=WORKING_DIR, capture_output=True, text=True)
+
+    push = ["git", "push", "-f", "origin", head]
+    result = subprocess.run(push, cwd=WORKING_DIR, capture_output=True, text=True)
+
+    if result.returncode != 0:
+        logger.error(f"Error pushing changes to branch {head}: {result.stderr}")
+        exit(1)
+
+
+def create_pr(repo: str, base: str, head: str, conflict: bool):
+    logger.info("Creating PR ...")
+
+    title = f"chore: merging diff into branch {base}"
+    body = f"Automated change: merging diff into branch {base}"
+
+    if conflict:
+        title = f"{title} CONFLICTS!"
+        body = f"**Conflicts! Need human intervention!**\n\n{body}"
+
+    repo = gh_client.get_repo(repo)
+    pr = repo.create_pull(base=base, head=head, title=title, body=body)
+    if not pr:
+        logger.error("Error creating PR.")
+        exit(1)
+    logger.info(f"PR {pr.number} created.")
+
+
+def get_all_chapters_branches(repo: str) -> typing.List[str]:
+    return sorted(
+        [
+            b.name
+            for b in gh_client.get_repo(repo).get_branches()
+            if re.search(r"\d", b.name)
+        ]
+    )
+
+
+def get_pr_base_branch(repo: str, pull_request_number: int) -> str:
+    return gh_client.get_repo(repo).get_pull(pull_request_number).base.ref
+
+
+def apply_diff_to_branch_and_create_pr(
+    repo: str, base: str, owner: str, ignore_conflicts: bool
+):
+    logger.info(f"=== Working on branch {base} ===")
+
+    head_branch = create_new_branch_based_on(base)
+
+    conflict = apply_patch()
+    if conflict and not ignore_conflicts:
+        return True
+
+    commit_diff(head_branch)
+
+    create_pr(repo, base, f"{owner}:{head_branch}", conflict)
+
+
+def cleanup():
+    logger.info("Cleaning up ...")
+
+    if os.path.exists(DIFF_FILE):
+        os.remove(DIFF_FILE)
+
+    shutil.rmtree(WORKING_DIR, ignore_errors=True)
+
+
+def main(
+    upstream_owner,
+    upstream_repo_name,
+    fork_owner,
+    fork_repo_name,
+    pull_request_number,
+    ignore_conflicts,
+):
+    upstream_repo = f"{upstream_owner}/{upstream_repo_name}"
+    fork_repo = f"{fork_owner}/{fork_repo_name}"
+
+    get_diff_as_patch(upstream_repo, pull_request_number)
+    clone(fork_repo)
+
+    pr_base = get_pr_base_branch(upstream_repo, pull_request_number)
+    target_branches = [
+        b for b in get_all_chapters_branches(upstream_repo) if b > pr_base
+    ]
+
+    for branch in target_branches:
+        conflict = apply_diff_to_branch_and_create_pr(
+            upstream_repo, branch, fork_owner, ignore_conflicts
+        )
+
+        if conflict and not ignore_conflicts:
+            logger.info(
```

Comment:
I would say that this is more of an `error` (or even `criticial`) than an `info`, ie. it's saying that the script cannot continue. I can see an argument that it's actually working as expected (go as far as you can and then stop), so if you feel `info` is more appropriate, feel free to just resolve/ignore this.